### DISCUSSION TITLE
how to get scan target from burp to dirsearch at real time?

### DISCUSSION THREAD

**Discussion by Tsunamori** [maurosoria/dirsearch#1095] [Q&A] [answered]
I've searched around from google but got nothing, is there any way to active the dirsearch and keep getting the scan target(s) from the burp? I mean, it looks like I could set a burp upstream proxy for dirsearch to get url input maybe, but the dirsearch seems not agree to get the url from burp, neither running constantly for multi times input.
So, is there any commands or extensions can solve that?

**Comment by shelld3v** [comment, marked as answer]
This has already been suggested in issue #159, but there are things need to be done before implementing a burp extension for dirsearch

**Reply by Tsunamori** [reply]
I see, thanks for the answer! I thought it would be sth like [xray](https://github.com/chaitin/xray), xray has a sample command `xray webscan --listen 127.0.0.1:7777 --html-output proxy.html`, when you set the [burp upstream proxy ](https://portswigger.net/support/burp-suite-upstream-proxy-servers) as _127.0.0.1:7777_ , xray will get all the targets from burp and scan it. The idea is to scan out sth like 'xxx.com/customizedname/xx/swagger'. But yes it would be better implementing a burp extension so we could add some regex or fuzz to filter and so on.
